### PR TITLE
HPCC-24967 Add chart annotations based on chksum of configmap

### PR DESCRIPTION
**Pull request by jakesmith**
This will force the pods to recycle, if a config property
relevant to a deployment changes.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@richardkchapman - if you could give this an initial review/feedback. Thanks.

File: helm/hpcc/templates/_helpers.tpl [hunk 1]
```diff
@@ -406,6 +406,12 @@ Check that the storage and spill planes for a component exist
 {{- end }}
 {{- end -}}
 
+{{/*
```
Comment (richardkchapman) [inline helm/hpcc/templates/_helpers.tpl:409]: This seems (a) a bit small to be worth having a template for and (b) to be unused

**Review by richardkchapman — CHANGES_REQUESTED** [review]

File: helm/hpcc/templates/_helpers.tpl (hunk 1 above)
Comment (jakesmith) [reply helm/hpcc/templates/_helpers.tpl:409]: agree, will remove.

**Comment by jakesmith** [comment]
@richardkchapman - have squashed in the above change.

File: helm/hpcc/templates/esp.yaml [hunk 2]
```diff
@@ -1,5 +1,29 @@
...
+      services:
+{{ include "hpcc.generateConfigMapServices" .root | indent 6 }}
+{{ include "hpcc.generateVaultConfig" (dict "root" .root "categories" (list "storage" "esp" ) ) | indent 6 }}
+    global:
+{{ include "hpcc.generateGlobalConfigMap" .root | indent 6 }}
+{{- end -}}
+
 {{ range $.Values.esp -}}
 {{- if not .disabled -}}
+{{- $configSHA := include "hpcc.espConfigMap" (dict "root" $ "me" .) | sha256sum }}
```
Comment (richardkchapman) [inline helm/hpcc/templates/esp.yaml:26]: I wonder whether a $scope "= dict "root" $ "me" . at the top of some of these would be helpful, as it's passed to a LOT of templates

File: helm/hpcc/templates/eclagent.yaml [hunk 3]
```diff
@@ -1,5 +1,69 @@
...
+  name: {{ .me.name }}-configmap
+data:
+  {{ .me.name }}.yaml: |
+    version: 1.0
+    eclagent:
+{{ toYaml (omit .me "logging") | indent 6 }}
+{{- include "hpcc.generateLoggingConfig" (dict "root" .root "me" .me) | indent 6 }}
+    {{ $apptype }}:
+{{ toYaml (omit .me "logging") | indent 6 }}
+{{- include "hpcc.generateLoggingConfig" (dict "root" .root "me" .me) | indent 6 }}
```
Comment (richardkchapman) [inline helm/hpcc/templates/eclagent.yaml:18]: Isn't this dict just "." ?

**Comment by jakesmith** [comment]
@richardkchapman - have made changes to common up usages of root+me dict, and rebased/retargeted to master

**Review by richardkchapman — DISMISSED** [review]
Looks good - please squash

**Comment by jakesmith** [comment]
@richardkchapman - squashed.

**Review by richardkchapman — APPROVED** [review]
Approved

**Pull request merged**